Packages/com.unity.inputsystem/InputSystem/Plugins/XR/GenericXRDevice.cs

```diff
@@ -1,35 +1,66 @@
+#if ENABLE_VR
+using UnityEngine.InputSystem.Controls;
 using UnityEngine.InputSystem.XR.Haptics;
-using UnityEngine.InputSystem.Haptics;
 using UnityEngine.InputSystem.Layouts;
-#if UNITY_INPUT_SYSTEM_ENABLE_XR
+using UnityEngine.Scripting;
 using UnityEngine.XR;
-#endif
 
 namespace UnityEngine.InputSystem.XR
 {
     /// <summary>
     /// The base type of all XR head mounted displays.  This can help organize shared behaviour across all HMDs.
     /// </summary>
     [InputControlLayout(isGenericTypeOfDevice = true, displayName = "XR HMD")]
-    [Scripting.Preserve]
-    public class XRHMD : InputDevice
+    [Preserve]
+    public class XRHMD : TrackedDevice
     {
+        [InputControl]
+        [Preserve]
+        public Vector3Control leftEyePosition { get; private set; }
+        [InputControl]
+        [Preserve]
+        public QuaternionControl leftEyeRotation { get; private set; }
+
+        [InputControl]
+        [Preserve]
+        public Vector3Control rightEyePosition { get; private set; }
+        [InputControl]
+        [Preserve]
+        public QuaternionControl rightEyeRotation { get; private set; }
+        [InputControl]
+        [Preserve]
+        public Vector3Control centerEyePosition { get; private set; }
+        [InputControl]
+        [Preserve]
+        public QuaternionControl centerEyeRotation { get; private set; }
```

Comment:
@StayTalm
Users have been asking why no controls are available at the HMD level. I moved the tracking-related controls (`isTracked`, etc) all the way up to `TrackedDevice` (same for `XRController`) and pulled the left/right/center eye controls (position&rotation) up to the HMD level. Make sense?